### PR TITLE
Update fog to 1.28.0

### PR DESCRIPTION
**Pull request by eLobato**
To be merged after https://github.com/theforeman/foreman-packaging/pull/547

Should `foreman/foreman.spec` be modified too (compute), or we wait for a release?

### CODE REVIEW

File: rubygem-fog/rubygem-fog.spec [hunk 1]
```diff
@@ -27,7 +27,7 @@ Patch11: fog-no-serverlove.patch
 Requires: %{?scl_prefix}rubygems
 Requires: %{?scl_prefix}rubygem(fog-aws) >= 0
 Requires: %{?scl_prefix}rubygem(fog-aws) < 1
-Requires: %{?scl_prefix}rubygem(fog-core) >= 1.27.3
+Requires: %{?scl_prefix}rubygem(fog-core) >= 1.29.0
```
Comment (domcleal) [inline rubygem-fog/rubygem-fog.spec]: Still 1.27.3

File: rubygem-fog/rubygem-fog.spec [hunk 2]
```diff
@@ -46,7 +46,7 @@ BuildRequires: %{?scl_prefix}rubygems-devel
 BuildRequires: %{?scl_prefix}rubygems
 BuildRequires: %{?scl_prefix}rubygem(fog-aws) >= 0
 BuildRequires: %{?scl_prefix}rubygem(fog-aws) < 1
-BuildRequires: %{?scl_prefix}rubygem(fog-core) >= 1.27.3
+BuildRequires: %{?scl_prefix}rubygem(fog-core) >= 1.29.0
```
Comment (domcleal) [inline rubygem-fog/rubygem-fog.spec]: ditto

**Comment by domcleal** [comment]
fog 1.28.0 adds a dependency on fog-riakcs, which we don't need.  What I usually do is add a patch to strip it back out again.  These are maintained here, in a -simplify branch based off of a tag: https://github.com/theforeman/fog/commits/v1.27.0-simplify.  The current patches would also need rebasing onto v1.28.0.  Are you OK doing this, or shall I?

Regarding foreman.spec, that needs updating too, referencing the Foreman ticket.  The PR to Foreman itself should be merged first though.

**Comment by domcleal** [comment]
Thanks for v1.28.0-simpllify.  You should be able to use `git format-patch -1` on it to generate a patch which you can add to this RPM, just copy the other lines present.  If you had to resolve conflicts in any other patches when rebasing then you should just replace those patch files too.

**Comment by eLobato** [comment]
Cool, https://github.com/theforeman/foreman-packaging/pull/550 updates the foreman-compute spec to use the newer fog and fog-aws versions

**Comment by eLobato** [comment]
@domcleal Latest update includes the patch removing riakcs & leaving fog-core dependency at 1.27.3

**Comment by domcleal** [comment]
@eLobato you need to add a PatchX line, a %patchX line and lastly, a 'sed' line to remove the dependency from the gemspec

**Comment by eLobato** [comment]
@domcleal Done, sorry for having Jenkins rebuild the packages all the time and keeping you reloading the page!

**Comment by domcleal** [comment]
No problem, looks good now.  Go ahead on the core change if you like, I'll run a last test of these PRs in parallel.

**Comment by domcleal** [comment]
Thanks @eLobato!  Building now.

**Pull request merged**